crates/conformance-test-utils/src/madara.rs

```diff
@@ -0,0 +1,85 @@
+use eyre::Result;
+use serde_json::Value;
+use starknet::core::types::FieldElement;
+use starknet::core::utils::get_storage_var_address;
+
+use crate::Felt;
+
+pub const FEE_TOKEN_ADDRESS_HEX: &str = "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7";
+
+/// Generates the genesis storage data for pre-funding a Starknet address on Madara.
+///
+/// This function calculates the storage key for the balance of the ERC20 Fee Token
+/// contract using the provided Starknet address. The resulting JSON object represents the initial
+/// storage of the Fee Token contract, where the account associated with the Starknet address is
+/// pre-funded with the specified `amount`.
+///
+/// # Arguments
+///
+/// * `starknet_address` - The Starknet address to be funded
+/// * `amount` - The amount of funds to allocate to the Starknet account.
+///
+/// # Returns
+///
+/// * `Result<Value>` - A JSON object representing the initial storage of the Fee Token contract
+///   where the Starknet address has been pre-funded with the specified `amount`. Returns an error
+///   if the constant `FEE_TOKEN_ADDRESS_HEX` could not be converted to a `FieldElement` or if there
+///   was an error calculating the storage variable address.
+pub fn genesis_fund_starknet_address(starknet_address: FieldElement, amount: FieldElement) -> Result<Value> {
```

Comment:
do we really want to return a json value? why not keeping rust types everywhere and eventually lets serde do the job? especially because I think that on the madara side, serde will be implemented for the genesis type so it should work without doing ourselves any serde

Reply:
Ah I see what you mean, let me refactor this to use the types instead of returning this JSON value

Reply:
@ftupas suggest to use some type aliasing like
```
type ContractAddress = Felt;
type StorageKey = Felt;
type StorageValue = Felt;
```
and then you can return 
```
((ContractAddress, StorageKey), StorageValue)
```
Or something similar.

Reply:
I'm thinking to use these [types](https://github.com/keep-starknet-strange/madara/blob/main/crates/pallets/starknet/src/types.rs#L18) directly from Madara instead, do you think it's a good idea?

Reply:
Indeed even better

Reply:
Thanks @greged93 for reaching out to Madara team 🙏 So for now, they are not modifying `Felt252Wrapper` to serialize into hex, therefore we will create our own types instead like so

```Rust
type ContractAddress = Felt;
type StorageKey = Felt;
type StorageValue = Felt;
```

 and not use Madara's types until they change the serialization. I've created a commit [here](https://github.com/ftupas/kakarot-rpc/commit/b96cb203a99221183cbfc81725419e32756c0016) that we can use later on if ever